Hide Bubble Chat from Blocked Users

Stickmasterluke opened the pull request.

Stickmasterluke committed:
Bubble Chat Displays Important Stuff

hockeychaos commented on CoreScriptsRoot/CoreScripts/BubbleChat.lua:
```diff
@@ -29,15 +29,17 @@ local BILLBOARD_MAX_HEIGHT = 500
 local CchMaxChatMessageLength = 128 -- max chat message length, including null terminator and elipses.
 local CchMaxChatMessageLengthExclusive = CchMaxChatMessageLength - string.len(ELIPSES) - 1
+
+local NON_CORESCRIPT_MODE = false
```
have you tested this script to see if it currently works in a non core script mode?

BinaryResolved replied:
This would be very useful. Especially in big games with spammers, where you can still see Bubble Chat.

hockeychaos commented on CoreScriptsRoot/CoreScripts/BubbleChat.lua:
```diff
@@ -48,6 +50,13 @@ local BubbleColor = {	WHITE = "dub",
 
 
+if not NON_CORESCRIPT_MODE then
+	while PlayersService.LocalPlayer == nil do PlayersService.ChildAdded:wait() end		--eww
```
I think this code will work, but technically LocalPlayer is property, so it isn't necessary for a child to be added for this property to change.

hockeychaos commented on CoreScriptsRoot/CoreScripts/BubbleChat.lua:
```diff
@@ -48,6 +50,13 @@ local BubbleColor = {	WHITE = "dub",
+	while PlayersService.LocalPlayer == nil do PlayersService.ChildAdded:wait() end		--eww
+	local GuiRoot = CoreGuiService:WaitForChild('RobloxGui')
+	playerDropDownModule = require(GuiRoot.Modules:WaitForChild("PlayerDropDown"))
+	blockingUtility = playerDropDownModule:CreateBlockingUtility()
```
this does not have any sort of local scope, should have this

hockeychaos commented on CoreScriptsRoot/CoreScripts/BubbleChat.lua:
```diff
@@ -552,12 +561,17 @@ local function createChatOutput()
-		-- eliminate display of emotes
-		if string.find(message, "/e ") == 1 or string.find(message, "/emote ") == 1 then return end
+		-- eliminate display of commands
+		if string.sub(message, 1, 1) == '/' then return end
```
all commands use a forward slash?

Stickmasterluke committed:
Fixes

Removed NON_CORESCRIPT_MODE variable because it implied that it could
run as a non-corescript, which is not true. Also implemented a check to
see if message text would actually be displayed after passing through.
Bubble read distance raised.

Stickmasterluke committed:
Reuseable TextLabel
